File: src/main/java/org/lucidfox/questfiller/model/npc/SoldItem.java
package org.lucidfox.questfiller.model.npc;

public final class SoldItem {
	private final String name;
	private final int price;

	public SoldItem(String name, int price) {
		this.name = name;
		this.price = price;
	}

	public String getName() {
		return name;
	}

	public int getPrice() {
		return price;
	}
	
	private static Integer getNonzeroOrNull(final int quantity) {
		return quantity == 0 ? null : quantity;
	}
	
	public Integer getGold() {
		return getNonzeroOrNull(price / 10000); 
	}
	
	public Integer getSilver() {
		return getNonzeroOrNull((price % 10000) / 100); 
	}
	
	public Integer getCopper() {
		return getNonzeroOrNull(price % 100); 
	}
}

File: src/model/npc/SoldItem.js
export default class SoldItem {
	/**
	 * 
	 * @param {string} name 
	 * @param {number} price 
	 * @param {number} quantity
	 */
	constructor(name, price, quantity) {
		this.name = name;
		this.price = price || 0;
		this.quantity = quantity === -1 ? null : (quantity || null);
	}

	getGold() {
		return Math.floor(this.price / 10000) || "";
	}

	getSilver() {
		return (Math.floor(this.price / 100) % 100) || "";
	}

	getCopper() {
		return (this.price % 100) || "";
	}

	/**
	 * 
	 * @param {SoldItem} other 
	 */
	compareTo(other) {
		return this.name.localeCompare(other.name);
	}
}
